client/dashboard/plugins/manage/components/plugin-switcher.tsx

```diff
@@ -20,12 +21,60 @@ export const PluginSwitcher = ( {
 	searchableFields,
 	selectedPluginSlug = '',
 	view,
+	setView,
+	paginationInfo,
 }: {
 	pluginsWithIcon: PluginListRow[];
 	searchableFields: Field< PluginListRow >[];
 	selectedPluginSlug?: string;
 	view: View;
+	setView: Dispatch< SetStateAction< View > >;
+	paginationInfo: { totalItems: number; totalPages: number };
 } ) => {
+	const scrollRef = useRef< HTMLDivElement >( null );
+
+	// Load next page when scrolling near bottom
+	const handleLoadMore = useCallback( () => {
+		if ( ! paginationInfo ) {
+			return;
+		}
+
+		const currentPage = view.page || 1;
+		const currentPerPage = view.perPage || 10;
+
+		if ( currentPage < paginationInfo.totalPages ) {
+			setView( ( currentView ) => ( {
+				...currentView,
+				perPage: ( currentPage + 1 ) * currentPerPage, // Accumulate items
```

Comment:
I think there's logic error here. We never update the current page, so this expression is always `2 * currentPerPage`, which basically doubles the page size every time we scroll.

I think what we want is loading the next page data by increasing the page size window. Something like:

```js
setView( currentView => ({
    ...currentView,
    perPage: view.perPage + DEFAULT_VIEW.perPage;
});
```